dvc/remote/local/__init__.py

```diff
@@ -566,3 +593,83 @@ def unprotect(path_info):
     @staticmethod
     def protect(path_info):
         os.chmod(path_info.path, stat.S_IREAD | stat.S_IRGRP | stat.S_IROTH)
+
+    @classmethod
+    def _append_unpacked_suffix(cls, string):
+        return string + cls.UNPACKED_DIR_SUFFIX
+
+    def _get_unpacked_dir_info(self, checksum):
+        unpacked_dir_info = self.checksum_to_path_info(checksum)
+        unpacked_dir_info.path = self._append_unpacked_suffix(
+            unpacked_dir_info.path
+        )
+
+        return unpacked_dir_info
+
+    def _update_unpacked_dir(self, checksum, dir_info):
+        unpacked_dir_info = self._get_unpacked_dir_info(checksum)
+
+        exists = self.exists(unpacked_dir_info)
+        if exists:
+            if self.state.get(unpacked_dir_info):
+                return
+
+            shutil.rmtree(unpacked_dir_info.path)
+
+        self._create_unpacked_dir(checksum, dir_info, unpacked_dir_info)
+
+    def _create_unpacked_dir(self, checksum, dir_info, unpacked_dir_info):
+        self.makedirs(unpacked_dir_info)
+        for entry in dir_info:
+            entry_cache_info = self.checksum_to_path_info(
+                entry[self.PARAM_CHECKSUM]
+            )
+            relpath = entry[self.PARAM_RELPATH]
+
+            unpacked_entry_info = copy(entry_cache_info)
+            unpacked_entry_info.path = self.ospath.join(
+                unpacked_dir_info.path, relpath
+            )
+            self.link(entry_cache_info, unpacked_entry_info, "hardlink")
+        self.state.save(unpacked_dir_info, checksum)
+
+    def _path_metadata_changed(self, path):
```

Comment:
Do we need this method at all? You've used state.get above

Reply:
It is not necessary. Removed